src/app/clusters/ota-requestor/OTARequestor.h

```diff
@@ -269,9 +265,9 @@ class OTARequestor : public OTARequestorInterface, public BDXDownloader::StateDe
     static void OnNotifyUpdateAppliedFailure(void * context, EmberAfStatus);
 
     OTARequestorDriver * mOtaRequestorDriver  = nullptr;
-    NodeId mProviderNodeId                    = kUndefinedNodeId;
-    FabricIndex mProviderFabricIndex          = kUndefinedFabricIndex;
-    EndpointId mProviderEndpointId            = kRootEndpointId;
+    NodeId mProviderNodeId                    = kUndefinedNodeId;      // Only valid for the current update in progress
```

Comment:
I'm actually unclear on what this comment is meant to tell me. It's not like we regularly update these values so it should be valid literally forever unless some code explicitly sets it to undefined again, no?

Reply:
The code added in this PR explicitly sets it to undefined. And I think that we should update these values regularly: every time DefaultProvider changes or AnnounceOTAProvider is received.

Reply:
Right... but the point is, once they have been set (either via DefaultOTAProviders change or AnnounceOTAProviders), then these contain "valid" values unless they provider node is no longer on the fabric.

Reply:
Here valid is supposed to mean "valid as the provider we are currently communicating with". If we get a fresh AnnounceOtaProvider then the mProviderNodeId gets overwritten, the old one is no longer valid. Same for CancelImageUpdate().